components/scream/cmake/Scorpio.cmake

```diff
@@ -65,46 +10,38 @@ macro (CreateScorpioTarget CLIB FLIB)
-  if (${CLIB} AND NOT TARGET scorpio_c)
+  if (NOT TARGET scream_pioc)
     # Get GPTL as a target
     include (GPTL)
     CreateGPTLTarget()
 
     # Get Netcdf libs
+    include (GetNetcdfLibs)
     GetNetcdfLibs()
 
     # Look for pioc lib in the lib subdirectory of the one stored in INSTALL_SHAREDPATH (set by CIME)
     find_library(SCORPIO_C_LIB pioc REQUIRED PATHS ${INSTALL_SHAREDPATH}/lib)
 
     # Create the interface library that scream targets can link to
-    add_library(scorpio_c INTERFACE IMPORTED GLOBAL)
-    # set_target_properties (scorpio_c PROPERTIES IMPORTED_LOCATION ${SCORPIO_C_LIB})
-    target_link_libraries(scorpio_c INTERFACE "${SCORPIO_C_LIB};scream_gptl;${netcdf_c_lib}")
+    add_library(scream_pioc UNKNOWN IMPORTED GLOBAL)
+    set_target_properties(scream_pioc PROPERTIES IMPORTED_LOCATION "${SCORPIO_C_LIB}")
+    target_link_libraries(scream_pioc INTERFACE "scream_gptl;${netcdf_c_lib}")
+    set_target_properties(scream_pioc PROPERTIES INTERFACE_INCLUDE_DIRECTORIES ${INSTALL_SHAREDPATH}/include)
     if (pnetcdf_lib)
-      target_link_libraries(scorpio_c INTERFACE "${pnetcdf_lib}")
+      target_link_libraries(scream_pioc INTERFACE "${pnetcdf_lib}")
     endif ()
+  endif ()
 
-    # Update the list of scream tpls
-    list(APPEND SCREAM_TPL_LIBRARIES scorpio_c)
-    list(APPEND SCREAM_TPL_INCLUDE_DIRS ${INSTALL_SHAREDPATH}/include)
-
-    # If f lib is requested (and we didn't already parsed this script), create interface lib
-    if (${FLIB} AND NOT TARGET scorpio_f)
-      # Look for piof lib in the lib subdirectory of the one stored in INSTALL_SHAREDPATH (set by CIME)
-      find_library(SCORPIO_F_LIB piof REQUIRED PATHS ${INSTALL_SHAREDPATH}/lib)
-
-      # Create the interface library that scream targets can link to
-      add_library(scorpio_f INTERFACE IMPORTED GLOBAL)
-      # set_target_properties (scorpio_f PROPERTIES IMPORTED_LOCATION ${SCORPIO_F_LIB})
-      target_link_libraries(scorpio_f INTERFACE ${SCORPIO_F_LIB} ${netecdf_f_lib} ${SCORPIO_C_LIB})
-      if (pnetcdf_lib)
-        target_link_libraries(scorpio_f INTERFACE "${pnetcdf_lib}")
-      endif ()
+  # If f lib is requested (and we didn't already parsed this script), create interface lib
+  if (${CREATE_FLIB} AND NOT TARGET scream_piof)
+    # Look for piof lib in the lib subdirectory of the one stored in INSTALL_SHAREDPATH (set by CIME)
+    find_library(SCORPIO_F_LIB piof REQUIRED PATHS ${INSTALL_SHAREDPATH}/lib)
 
-      # Update the list of scream tpls
-      list(APPEND SCREAM_TPL_LIBRARIES scorpio_f)
-      list(APPEND SCREAM_TPL_INCLUDE_DIRS ${INSTALL_SHAREDPATH}/include)
-    endif ()
+    # Create the imported library that scream targets can link to
+    add_library(scream_piof UNKNOWN IMPORTED GLOBAL)
+    set_target_properties(scream_piof PROPERTIES IMPORTED_LOCATION "${SCORPIO_F_LIB}")
+    target_link_libraries(scream_piof INTERFACE "${netecdf_f_lib};scream_pioc")
```

Comment:
Is `netecdf_f_lib` a typo? I would have expected `netcdf_f_lib`

Reply:
Great catch!